src/plugins/drawing/objects/text.tsx

```diff
@@ -0,0 +1,203 @@
+  IToolbarButtonProps, typeField } from "./drawing-object";
+import { BoundingBoxDelta, Point, ToolbarSettings } from "../model/drawing-basic-types";
+import TextToolIcon from "../../../assets/icons/comment/comment.svg";
+import { SvgToolModeButton } from "../components/drawing-toolbar-buttons";
+import { uniqueId } from "../../../../src/utilities/js-utils";
+import { WrappedSvgText } from "../components/wrapped-svg-text";
+
+export const TextObject = EditableObject.named("TextObject")
+  .props({
+    type: typeField("text"),
+    width: types.number,
+    height: types.number,
+    stroke: types.string,
+    text: types.string
+  })
+  .volatile(self => ({
+    dragWidth: undefined as number | undefined,
+    dragHeight: undefined as number | undefined
+  }))
+  .views(self => ({
+    get currentDims() {
+      const { width, height, dragWidth, dragHeight } = self;
+      return {
+        width: dragWidth ?? width,
+        height: dragHeight ?? height
+      };
+    }
+  }))
+  .views(self => ({
+    get boundingBox() {
+      const { x, y } = self.position;
+      const { width, height } = self.currentDims;
+      const nw: Point = {x, y};
+      const se: Point = {x: x + width, y: y + height};
+      return {nw, se};
+    }
+  }))
+  .actions(self => ({
+    setStroke(stroke: string){ 
+      self.stroke = stroke; 
+    },
+    setText(text: string) {
+      self.text = text;
+    },
+    resize(start: Point, end: Point) {
+      self.x = Math.min(start.x, end.x);
+      self.y = Math.min(start.y, end.y);
+      self.width = Math.max(start.x, end.x) - self.x;
+      self.height = Math.max(start.y, end.y) - self.y;
+    },
+    setDragBounds(deltas: BoundingBoxDelta) {
+      self.dragX = self.x + deltas.left;
+      self.dragY = self.y + deltas.top;
+      self.dragWidth  = self.width  + deltas.right - deltas.left;
+      self.dragHeight = self.height + deltas.bottom - deltas.top;
+    },
+    adoptDragBounds() {
+      self.adoptDragPosition();
+      self.width = self.dragWidth ?? self.width;
+      self.height = self.dragHeight ?? self.height;
+      self.dragWidth = self.dragHeight = undefined;
+    },
+    setEditing(editing: boolean) {
+      console.log('Editing state is now', editing);
+      self.isEditing = editing;
+    }
+  }));  
+
+export interface TextObjectType extends Instance<typeof TextObject> {}
+export interface TextObjectSnapshot extends SnapshotIn<typeof TextObject> {}
+
+export function isTextObject(model: DrawingObjectType): model is TextObjectType {
+  return model.type === "text";
+}
+
+export const TextComponent = observer(
+    function TextComponent({model, handleHover, handleDrag} : IDrawingComponentProps) {
+  const textEditor = useRef<HTMLTextAreaElement>(null);
+  if (!isTextObject(model)) return null;
+  const textobj = model as TextObjectType;
+  const { id, stroke, text } = textobj;
+  const { x, y } = model.position;
+  const { width, height } = textobj.currentDims;
+  const clipId = uniqueId();
```

Comment:
Does it matter that this changes on every render?

Reply:
It doesn't matter what the ID is, but I'm open to a more elegant solution if you have one. Using a fixed value does not work, because there can be multiple text components in the drawing.